src/v/cluster/metrics_reporter.cc

```diff
@@ -312,10 +313,14 @@ ss::future<> metrics_reporter::try_initialize_cluster_info() {
 
     auto& first_cfg = batches.front();
 
+    _cluster_info.creation_timestamp = first_cfg.header().first_timestamp;
+    co_await _feature_table.invoke_on_all([&](features::feature_table& ft) {
+        ft.set_builtin_trial_license(_cluster_info.creation_timestamp);
```

Comment:
what do we do if creation timestamp is bogus (e.g. ntp isn't running and clocks aren't accurate)?

Reply:
Yeah, that's a good point. Specifically, when only the creation timestamp is bogus, but now the time is accurate, right? Surely, all bets are off if the time is bogus throughout the cluster's runtime (e.g., license checks against the timestamp in the license would break equally). Do you have any ideas? The only improvement I can think of is sanity-checking that the creation timestamp is in the past, but it's hard to determine the cluster creation time retrospectively because the prefix of the controller log is snapshotted away after ~60s.